orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -61,6 +61,18 @@ def __init__(self, module):
         # TrainingAgent or InferenceAgent
         self._execution_agent = None
 
+        # enable fast path to skip some repeated logic if user knows that there will be no change for the entire training
+        # TODO - move it to config API
+        self._freeze = False
+        # enabled only when _freeze is True and after execution session is created the first time
+        # when _fast_path is enabled, these checks will be turned off:
+        # - schema check in _export_model() to determine whether to build gradient graph again
+        #   -- since there is no need to rebuild gradient graph, the following two functions can be skipped as well:
+        #      parse_inputs_for_onnx_export() and _reinitialize_graph_builder()
+        # - logic to determine whether to create execution session again
+        # - further checks on device change in both forward and backward pass
+        self._fast_path = False
```

Comment:
These two flag names are too abstract and it is not possible to know what is being frozen and what is the fast path. Use a more intuitive names

For self._freeze, it is not clear what it "freezes" even after reading the comment. Elaborate on that

Does self._fast_path skip input schema? output schema? both ? If both, `self._skip_io_schema` would be more intuitive. There can be different "fast path" implementations that skip different checks

Reply:
If more than one skip can happen, maybe we should create different flags. Or maybe a `dataclass class

```
from dataclasses import dataclass

@dataclass
class SkipCheck():
    skip_input_schema: bool
    skip_output_schema: bool
    skip_device_check: bool
```

can be used to group all skipping that can take place. This class can be hard-coded at `GraphExecutionManager` constructor

Reply:
code refactored to be more intuitive